### PR TITLE
integrate the new audio HRTF in AudioMixer

### PR DESCRIPTION
**Pull request by birarda**
- leverage @kencooke's new HRTF in our AudioMixer, remove a bunch of now unneeded Audio classes

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 04dc55686b9c751605406353e0fac5db7059195b available at: https://gist.github.com/7e5a8147212559b9c306<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7053.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7053.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 84b87b6f381442e7d16369657f97ac32d9061ba8 available at: https://gist.github.com/7e5a8147212559b9c306<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7053.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7053.dmg

File: assignment-client/src/audio/AudioMixerClientData.cpp [hunk 1]
```diff
@@ -126,45 +153,38 @@ int AudioMixerClientData::parseData(ReceivedMessage& message) {
...
-    }
-}
+    auto it = _audioStreams.begin();
+    while (it != _audioStreams.end()) {
+        SharedStreamPointer stream = it->second;
 
-void AudioMixerClientData::removeDeadInjectedStreams() {
+        static const int INJECTOR_INACTIVITY_USECS = 5 * USECS_PER_SECOND;
 
-    const int INJECTOR_CONSECUTIVE_NOT_MIXED_AFTER_STARTED_THRESHOLD = 100;
```
Comment (PhilipRosedale) [inline assignment-client/src/audio/AudioMixerClientData.cpp]: What are the units here?  still usecs?

Comment (birarda) [inline assignment-client/src/audio/AudioMixerClientData.cpp]: that's an old constant I deleted - it was number of events

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2cecfc499bcb2fd8c4ce1b45f8336bab69f86a5c available at: https://gist.github.com/7e5a8147212559b9c306<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7053.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7053.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5c9f673e38c8995867aedae3d90ba6fea52bd783 available at: https://gist.github.com/7e5a8147212559b9c306<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7053.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7053.dmg

File: libraries/audio/src/AudioHRTF.cpp [hunk 2]
```diff
@@ -509,10 +509,10 @@ static void setAzimuthAndGain(float firCoef[4][HRTF_TAPS], float bqCoef[5][4], i
                               int index, float azimuth, float gain, int channel) {
 
     // convert from radians to table units
-    //azimuth *= HRTF_AZIMUTHS / (2.0f * M_PI);
+    azimuth *= HRTF_AZIMUTHS / (2.0f * M_PI);
 
     // convert from degrees to table units
-    azimuth *= HRTF_AZIMUTHS / 360.0f;
+//    azimuth *= HRTF_AZIMUTHS / 360.0f;
```
Comment (PhilipRosedale) [inline libraries/audio/src/AudioHRTF.cpp]: remove dup debug?

File: assignment-client/src/audio/AudioMixerClientData.cpp (hunk 1 above)
Comment (PhilipRosedale) [inline assignment-client/src/audio/AudioMixerClientData.cpp]: sorry read backwards - thanks for adding the right units!

**Comment by hifi-gustavo** [comment]
Build console output for commit: cafcb6b109090e77bfd4c617efbf7648b3c7d3a8 available at: https://gist.github.com/7e5a8147212559b9c306<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7053.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7053.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 87d1d0a49c605f4c2173cc5dadd71ab4c1686d9c available at: https://gist.github.com/7e5a8147212559b9c306<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7053.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7053.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: e0cb8db914e2dae4c944672a24c2f65648b271ff available at: https://gist.github.com/7e5a8147212559b9c306<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7053.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7053.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1a7738353afe5966fdc023668b24d3bc8357391d available at: https://gist.github.com/7e5a8147212559b9c306<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7053.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7053.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: a8841b475fe9263c2182339ac4906490f5f25feb available at: https://gist.github.com/7e5a8147212559b9c306<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7053.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7053.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 26fd0c305a0e43e958ffaa30fcbb5b5ff79988e1 available at: https://gist.github.com/7e5a8147212559b9c306<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7053.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7053.dmg

**Comment by birarda** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 26fd0c305a0e43e958ffaa30fcbb5b5ff79988e1 available at: https://gist.github.com/7e5a8147212559b9c306<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7053.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7053.dmg

**Comment by kencooke** [comment]
Tested Windows version.

**Pull request merged**